python/tank/api.py

```diff
@@ -123,21 +122,10 @@ def execute_core_hook_method(self, hook_name, method_name, **kwargs):
 
     def log_metric(self, action, log_once=False):
         """
-        Log a core metric.
-
-        :param action: Action string to log, e.g. 'Init'
-        :param bool log_once: ``True`` if this metric should be ignored if it
-            has already been logged. Defaults to ``False``.
-
-        Logs a user activity metric as performed within core. This is
-        a convenience method that auto-populates the module portion of
-        `tank.util.log_user_activity_metric()`
-
-        Internal Use Only - We provide no guarantees that this method
-        will be backwards compatible.
+        This method is now deprecated and shouldn't be used anymore.
```

Comment:
i am trying to understand the implication of this deprecation. @sgsteph what will change if a user upgrades to this version of core? This is an 'internal' method so it should never be called from outside of core, so I am guessing that this is a pure refactor right? A user upgrading to core (but leaving engines and apps as is will see no change, right?

Reply:
I suggested we deprecate this to be on the safe side just in case there are clients who might be calling it. Maybe we should just remove it?

Reply:
deprecation sounds right to me too - it doesn't hurt to keep it around to make sure nothing breaks.